### PR TITLE
Questionnaire in the sign-up flow

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

Create a questionnaire with 3 questions to a new customer.

### Why was this change needed?

We want to understand our customers better to boost sales and onboarding.
Also, we want to display appropriate use-case on the Get Started page to onboard more smoothly.

### Other information (Screenshots)

image

https://www.loom.com/share/3baad9ecebe04a3e80eaddad547acebc

### CODE REVIEW

File: apps/web/src/pages/auth/components/QuestionnaireForm.tsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+import { api } from '../../../api/api.client';
+import { useAuthContext } from '../../../components/providers/AuthProvider';
+import { useVercelIntegration, useVercelParams } from '../../../hooks';
+import { ROUTES } from '../../../constants/routes.enum';
+import { DynamicCheckBox } from './dynamic-checkbox/DynamicCheckBox';
+
+type Props = {};
+
+export function QuestionnaireForm({}: Props) {
```
Comment (ainouzgali) [inline apps/web/src/pages/auth/components/QuestionnaireForm.tsx:36]:
```suggestion
export function QuestionnaireForm() {
```

File: apps/web/src/pages/auth/components/QuestionnaireForm.tsx [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        render={({ field }) => {
+          return (
+            <Select
+              label="Job title"
+              data-test-id="questionnaire-job-title"
+              required
+              error={errors.jobTitle?.message}
+              {...register('jobTitle', {
+                required: 'Please input your job title',
+              })}
```
Comment (ainouzgali) [inline apps/web/src/pages/auth/components/QuestionnaireForm.tsx:120]: If you are using the Controller you don't need the register as well. For the error message you can use 
```
<Controller ...
rules={{
          required: 'Required - Job title',
        }}
render = ...

```

File: apps/web/src/pages/auth/components/QuestionnaireForm.tsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+              getCreateLabel={(newOrganization) => <div>+ Add "{newOrganization}"</div>}
```
Comment (ainouzgali) [inline apps/web/src/pages/auth/components/QuestionnaireForm.tsx:121]: Why is this needed? Isn't it just a dropdown from a pre defined set of jobs?

File: apps/web/src/pages/auth/components/QuestionnaireForm.tsx
Comment (ainouzgali) [inline apps/web/src/pages/auth/components/QuestionnaireForm.tsx:1]: @djabarovgeorge I feel a better practice would be to either go with Controller or register - and not a mix of them :)

File: apps/web/src/pages/auth/components/dynamic-checkbox/DynamicCheckBox.tsx
Comment (ainouzgali) [inline apps/web/src/pages/auth/components/dynamic-checkbox/DynamicCheckBox.tsx:1]: Maybe not now - but we could consider adding this to our design system

File: libs/design-system/src/icons/general/DelayAction.tsx [hunk 4]
```diff
@@ -25,7 +24,7 @@ export function DelayAction(props: React.ComponentPropsWithoutRef<'svg'>) {
       </mask>
       <g mask="url(#mask0_433_2481)">
         <path
-          fill="url(#paint0_linear_433_2481)"
```
Comment (ainouzgali) [inline libs/design-system/src/icons/general/DelayAction.tsx:28]: Can you please make sure this doesn't remove the gradient on the step in the workflow editor?
Screen Shot 2023-12-27 at 11 30 39

**Review by ainouzgali — COMMENTED** [review]
Looks really good @djabarovgeorge

File: libs/design-system/src/icons/general/DelayAction.tsx (hunk 4 above)
Comment (djabarovgeorge) [reply libs/design-system/src/icons/general/DelayAction.tsx:28]: You are I used the incorrect icon by mistake! :(

File: apps/web/src/pages/auth/components/QuestionnaireForm.tsx (hunk 3 above)
Comment (djabarovgeorge) [reply apps/web/src/pages/auth/components/QuestionnaireForm.tsx:121]: Good catch! we don't need it, like it is leftover 🧹

File: apps/web/src/pages/auth/components/QuestionnaireForm.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/auth/components/QuestionnaireForm.tsx:1]: Good point! looks like i made some mess while playing with the form 🤦‍♂️

**Review by ainouzgali — APPROVED** [review]
